lightning/src/ln/channelmanager.rs

```diff
@@ -9093,6 +9095,152 @@ where
 			self.finish_close_channel(failure);
 		}
 	}
+
+	/// Utility for creating a BOLT11 invoice that can be verified by [`ChannelManager`] without
+	/// storing any additional state. It achieves this by including a [`PaymentSecret`] in the
+	/// invoice for which it uses to verify that the invoice has not expired and the payment amount
+	/// is sufficient, reproducing the [`PaymentPreimage`] if applicable.
+	pub fn create_bolt11_invoice(
+		&self, params: Bolt11InvoiceParameters,
+	) -> Result<Bolt11Invoice, SignOrCreationError<()>> {
+		let Bolt11InvoiceParameters {
+			currency, amount_msats, description, invoice_expiry_delta_secs,
+			min_final_cltv_expiry_delta, payment_hash,
+		} = params;
+
+		let currency = match currency {
+			Some(currency) => currency,
+			None => Network::from_chain_hash(self.chain_hash).map(Into::into).unwrap_or(Currency::Bitcoin),
+		};
```

Comment:
I'm wondering whether it makes sense to support a currency other than the one corresponding to `ChannelManager::chain_hash`. It won't be receivable by this `ChannelManager`, IIUC. Kinda seems like a bug that it was supported previously?

Reply:
Dropping it seems reasonable. @TheBlueMatt Any objections? We'd have to drop it from the deprecated utils, too.

Reply:
Currency now removed.

`git diff d7e897c7e266d392149aaec7a99e3aa140af5be5..7878801be79928e0309c079ea117d33a2521b557`